neusomatic/python/generate_dataset.py

```diff
@@ -849,116 +849,176 @@ def find_records(input_record):
-                                        (record_[4]) == (record_[14]):
-                                    if (len(record_[3]) > len(record_[13]) and record_[3][0:len(record_[13])] == record_[13]) or (
-                                            len(record_[13]) > len(record_[3]) and record_[13][0:len(record_[3])] == record_[3]):
+                    if emit_flag:
+                        if curren_pos_records:
+                            rrs = []
+                            for record_ in curren_pos_records:
+                                chrom, pos, ref, alt = [str(record_[0]), int(
+                                    record_[1]), record_[3], record_[4]]
+                                r_ = []
+                                if len(ref) == len(alt) and len(ref) > 1:
+                                    for ii in range(len(ref)):
+                                        ref_ = ref[ii]
+                                        alt_ = alt[ii]
+                                        if ref_ != alt_:
+                                            r_.append(
+                                                [chrom, pos + ii, ref_, alt_])
+                                else:
+                                    r_ = [[chrom, pos, ref, alt]]
+
+                                ann = [0] * 93
+                                if record_[1] == record_[11]:
+                                    if record_[3] == record_[13] and record_[4] == record_[14]:
                                         ann = record_[15:]
-                                elif len(record_[3]) < len(record_[4]) and len(record_[13]) < len(record_[14]) and \
-                                        (record_[3]) == (record_[13]):
-                                    if (len(record_[4]) > len(record_[14]) and record_[4][0:len(record_[14])] == record_[14]) or (
-                                            len(record_[14]) > len(record_[4]) and record_[14][0:len(record_[4])] == record_[4]):
-                                        ann = record_[15:]
-                            if ann:
-                                ann = list(map(float, ann))
-                            rrs.append([r_, ann])
-                        max_ann = max(map(lambda x: sum(x[1]), rrs))
-                        if max_ann > 0:
-                            rrs = list(filter(lambda x: sum(x[1]) > 0, rrs))
-                        elif max_ann == 0:
-                            rrs = rrs[0:1]
-                        for r_, ann in rrs:
-                            for rr in r_:
-                                records.append(rr + [str(i)])
-                                anns[i] = ann
-                                i += 1
-                    emit_flag = False
-                    if j < len_in_ensemble_bed:
+                                    elif (len(record_[3]) > len(record_[4]) and len(record_[13]) > len(record_[14]) and
```

Comment:
The sequence of these expressions makes the code less readable. Is there a better way?